lib/Tools/circt-lec/ConstructLEC.cpp

```diff
@@ -142,14 +150,16 @@ void ConstructLECPass::runOnOperation() {
 
   // TODO: we should find a more elegant way of reporting the result than
   // already inserting some LLVM here
-  Value eqFormatString =
-      lookupOrCreateStringGlobal(builder, getOperation(), "c1 == c2\n");
-  Value neqFormatString =
-      lookupOrCreateStringGlobal(builder, getOperation(), "c1 != c2\n");
-  Value formatString = builder.create<LLVM::SelectOp>(
-      loc, areEquivalent, eqFormatString, neqFormatString);
-  builder.create<LLVM::CallOp>(loc, printfFunc.value(),
-                               ValueRange{formatString});
-
-  builder.create<func::ReturnOp>(loc, ValueRange{});
+  if (insertReporting) {
+    Value eqFormatString =
+        lookupOrCreateStringGlobal(builder, getOperation(), "c1 == c2\n");
+    Value neqFormatString =
+        lookupOrCreateStringGlobal(builder, getOperation(), "c1 != c2\n");
+    Value formatString = builder.create<LLVM::SelectOp>(
+        loc, areEquivalent, eqFormatString, neqFormatString);
+    builder.create<LLVM::CallOp>(loc, printfFunc.value(),
+                                 ValueRange{formatString});
+
+    builder.create<func::ReturnOp>(loc, ValueRange{});
+  }
```

Comment:
Nit: from a code-structure pov it might be better to factor out the common code between the two `insertReporting` if-statements into a separate ConstructLECPass member function and do something like
```
if (!insertReporting) {
  thatHelperFunction(args);
  return;
}

<code currently in first if-statement>
thatHelperFunction(args);
<code currently in second if-statement>
```